From 1b230b87b8df0b0b996589a3275b27bd1d6a9c33 Mon Sep 17 00:00:00 2001
From: Scott Hess <shess@chromium.org>
Date: Tue, 23 Dec 2014 13:22:46 -0800
Subject: [PATCH 11/23] [fts3][backport] ICU tokenizer iterate by UTF16.

This issue was discovered in Gears and is upstream at:
   http://www.sqlite.org/src/info/892b74116a

No original review URL because this was part of the initial Chromium commit.
---
 third_party/sqlite/src/ext/fts3/fts3_icu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/sqlite/src/ext/fts3/fts3_icu.c b/third_party/sqlite/src/ext/fts3/fts3_icu.c
index 85390d3..a75b14a 100644
--- a/third_party/sqlite/src/ext/fts3/fts3_icu.c
+++ b/third_party/sqlite/src/ext/fts3/fts3_icu.c
@@ -198,7 +198,7 @@ static int icuNext(
 
     while( iStart<iEnd ){
       int iWhite = iStart;
-      U8_NEXT(pCsr->aChar, iWhite, pCsr->nChar, c);
+      U16_NEXT(pCsr->aChar, iWhite, pCsr->nChar, c);
       if( u_isspace(c) ){
         iStart = iWhite;
       }else{
-- 
2.2.1

